bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -972,6 +982,51 @@ void notify(@NonNull Error error,
         }
     }
 
+    /**
+     * Reports an error that occurred within the notifier to bugsnag. A lean error report will be
+     * generated and sent asynchronously with no callbacks, retry attempts, or writing to disk.
+     * This is intended for internal use only, and reports will not be visible to end-users.
+     */
+    void reportInternalBugsnagError(@NonNull Error error) {
+        error.setAppData(appData.getAppDataSummary());
+        error.setDeviceData(deviceData.getDeviceDataSummary());
+
+        MetaData metaData = error.getMetaData();
+        Notifier notifier = Notifier.getInstance();
+        metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "notifierName", notifier.getName());
+        metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "notifierVersion", notifier.getVersion());
+        metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "apiKey", config.getApiKey());
+
+        Object packageName = appData.getAppData().get("packageName");
+        metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "packageName", packageName);
+
+        final Report report = new Report(null, error);
+        try {
+            Async.run(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Delivery delivery = config.getDelivery();
+
+                        // can only modify headers if DefaultDelivery is in use
+                        if (delivery instanceof DefaultDelivery) {
+                            Map<String, String> headers = config.getErrorApiHeaders();
+                            headers.put("Bugsnag-Internal-Error", "true");
+                            headers.remove(Configuration.HEADER_API_KEY);
```

Comment:
I cant think why this needs to be removed?

Reply:
The original proposal was to remove the API key rather than add in the `Bugsnag-Internal-Error` header - I'll update so that we don't remove the API key.

Reply:
We don't send the API key in the payload or header because if on-prem users upgrade their notifier without upgrading on-prem, these internal error reports will be delivered to their dashboard.